requirements/requirements.txt

```diff
@@ -21,3 +21,6 @@ matplotlib
 pandas>=0.22
 scikit-learn>=0.20
 GitPython>=3.1.1
+
+# OmegaConf
+omegaconf
```

Comment:
is it possible to make it optional dependency?

Reply:
No. In current realisation all confguration "*.yaml" files are parsed and merged with OmegaConf.
If we want to leave catalyst-config functionality as it is and provide new functionality independently and make it optional I think we should create independent scripts `omegaconf-run.py` and `omegaconf-parser.py` for example.

Reply:
I think if you create alternative entry points you should use Hydra for it and not invent your own mechanism.

Reply:
Agree. Hydra encapsulates configuration parsing. It's convenient.

Reply:
@Scitator,
The original logic with merge_dicts below shows that catalyst is already solving a subset of the problems Hydra is solving (It is a form of config composition).
Large ML frameworks are switching to Hydra for it's config composition capabilities (recent example is fairseq).
Doing it in a way that retains backward compatibility is hard but possible, at least judging based on their experience.

I think embracing Hydra is a strategic project decision. In the short term using OmegaConf as the underlying config abstraction will be beneficial and will make it easier to interface with Hydra if and when the decision happens.

Some of the benefits:
* Across the codebase, you can use `cfg.foo.bar` instead of `cfg["foo"]["bar"]`
* Usage of interpolation (think of them as pointers in the config).
* Rudimentary command line integration (Hydra has a similar but more powerful CLI).
* More robust config merging capabilities
* Type safe configs via dataclasses (see Structured Configs in the OmegaConf docs).

Reply:
firstly, I don't think we need new entry point, we could just add a new argument, like `--hydra-config` (or something like this)
for example, during parser creation you could check `if IS_HYDRA_CONFIG_AVAILABLE` (more examples here - https://github.com/catalyst-team/catalyst/blob/master/catalyst/tools/settings.py) and add extra params and their parsing logic. this solution looks good for me, because it provides extra features without breaking changes

Catalyst Config API **must** save it backward compatibility - it's our production-ready solution :)

Reply:
@omry could you please provide more examples?
for example, I am not sure, that ``cfg.foo.bar instead of cfg["foo"]["bar"]`` is better solution - I personally prefer the second variant because it shows, that you are working with the config

other features looks great, but it examples are much better :)